pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -224,115 +214,68 @@ func (rq *ResourceQuotaController) syncResourceQuotaFromKey(key string) (err err
 	return rq.syncResourceQuota(quota)
 }
 
-// syncResourceQuota runs a complete sync of current status
-func (rq *ResourceQuotaController) syncResourceQuota(quota api.ResourceQuota) (err error) {
-
+// syncResourceQuota runs a complete sync of resource quota status across all known kinds
+func (rq *ResourceQuotaController) syncResourceQuota(resourceQuota api.ResourceQuota) (err error) {
 	// quota is dirty if any part of spec hard limits differs from the status hard limits
-	dirty := !api.Semantic.DeepEqual(quota.Spec.Hard, quota.Status.Hard)
+	dirty := !api.Semantic.DeepEqual(resourceQuota.Spec.Hard, resourceQuota.Status.Hard)
 
 	// dirty tracks if the usage status differs from the previous sync,
 	// if so, we send a new usage with latest status
 	// if this is our first sync, it will be dirty by default, since we need track usage
-	dirty = dirty || (quota.Status.Hard == nil || quota.Status.Used == nil)
+	dirty = dirty || (resourceQuota.Status.Hard == nil || resourceQuota.Status.Used == nil)
 
-	// Create a usage object that is based on the quota resource version
+	// Create a usage object that is based on the quota resource version that will handle updates
+	// by default, we preserve the past usage observation, and set hard to the current spec
+	previousUsed := api.ResourceList{}
+	if resourceQuota.Status.Used == nil {
```

Comment:
I believe it should be `if resourceQuota.Status.Used != nil {`. But since it works even with `Used` being `nil`, I would ditch the condition completely.